src/main/java/nus/climods/logic/parser/AddCommandParser.java

```diff
@@ -0,0 +1,36 @@
+package nus.climods.logic.parser;
+
+import java.util.stream.Stream;
+
+import nus.climods.logic.commands.AddCommand;
+import nus.climods.logic.parser.exceptions.ParseException;
+import nus.climods.logic.parser.parameters.ModuleCodeParameter;
+
+
+/**
+ * Parses input arguments and creates a new AddCommand object
+ */
+public class AddCommandParser implements Parser<AddCommand> {
+
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given {@code
+     * ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddCommand and returns an AddCommand object
+     * for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddCommand parse(String args) throws ParseException {
+        ModuleCodeParameter mcp = new ModuleCodeParameter(args);
```

Comment:
Right now the module code is case sensitive, should we make it case insensitive?

Reply:
agree, will edit accordingly